shell/platform/darwin/common/availability_version_check.cc

```diff
@@ -36,13 +145,50 @@ void InitializeAvailabilityCheck(void* unused) {
   }
   AvailabilityVersionCheck = reinterpret_cast<AvailabilityVersionCheckFn>(
       dlsym(RTLD_DEFAULT, "_availability_version_check"));
-  FML_CHECK(AvailabilityVersionCheck);
+  if (AvailabilityVersionCheck) {
+    return;
+  }
+
+  // If _availability_version_check can't be dynamically loaded, then version
+  // information must be parsed out of a system plist file.
+  auto product_version = flutter::ProductVersionFromSystemVersionPList();
+  if (product_version.has_value()) {
+    g_version = product_version.value();
+  } else {
+    // If reading version info out of the system plist file fails, then
+    // fall back to the minimum version that Flutter supports.
+    g_version.major = 11;
+    g_version.minor = 0;
+    g_version.subminor = 0;
+  }
 }
 
 extern "C" bool _availability_version_check(uint32_t count,
                                             dyld_build_version_t versions[]) {
   dispatch_once_f(&DispatchOnceCounter, NULL, InitializeAvailabilityCheck);
-  return AvailabilityVersionCheck(count, versions);
+  if (AvailabilityVersionCheck) {
```

Comment:
Don't you need to verify that the count is at least 1 so as not to read past the end of the array?

Reply:
I think we can assume that the implementation from the system does appropriate checking, so I added a check below before the emulated implementation.